waiter/src/waiter/scheduler/marathon.clj

```diff
@@ -275,7 +275,9 @@
      :cpus cpus
      :healthChecks [{:protocol (-> (or health-check-proto backend-proto) hu/backend-proto->scheme str/upper-case)
                      :path health-check-url
-                     :gracePeriodSeconds grace-period-secs
+                     :gracePeriodSeconds (if (pos? grace-period-secs)
+                                           grace-period-secs
+                                           (-> instance-expiry-mins t/minutes t/in-seconds (* 2)))
```

Comment:
Nitpick: Should we add a comment explaining why we're setting it like this? E.g., "Marathon's gracePeriodSeconds has a default value of 300. Using 2x the instance expiry limit is effectively equivalent to an infinite grace period."

Reply:
added comment